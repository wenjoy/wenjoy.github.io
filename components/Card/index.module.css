.container {
  margin: 0;
}

.frontface {
  background: #e4f0d0;
  user-select: none;
  cursor: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' width='40' height='48' viewport='0 0 100 100' style='fill:black;font-size:24px;'><text y='50%'>âœ‹</text></svg>") 16 0, auto;
}

.frontface:hover {
  box-shadow: 0px 0px 5px lightgray;
  margin-top: -5px;
}

.backface {
  background-color: gray;
background-image: repeating-linear-gradient(45deg, transparent, transparent 15px, rgba(255,255,255,.5) 15px, rgba(255,255,255,.5) 30px);
}

.card {
  min-width: 100%;
  min-height: 80%;
  border: 1px solid lightgray;
  border-radius: 3px;
  padding: 5px;
  line-height: 2;
  text-align: center;
}

.owner {
  min-width: 100%;
  min-height: 20%;
  font-size: 16px;
  white-space: nowrap;
  overflow: hidden;
  text-align: center;
  text-overflow: ellipsis;
}

.small {
  width: 60px;
  height: 80px;
  font-size: 30px;
}

.large {
  width: 100px;
  height: 180px;
}