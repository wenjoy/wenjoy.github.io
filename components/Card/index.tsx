interface ICardProps {
    point?: number | string
}
export default function Card({ point = 0 } : ICardProps) {
    return (
        <div className="card">
            {point}
        </div>
    )
}