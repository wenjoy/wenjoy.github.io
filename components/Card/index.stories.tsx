import { ComponentStory, ComponentMeta } from '@storybook/react';
import Card, {Point} from '.'
export default {
  title: 'Example/Card',
  component: Card,
} as ComponentMeta<typeof Card>;


const clickHandler = (point: Point) => {
  alert('you clicked ' + point)
}

const Template: ComponentStory<typeof Card> = (args) => <Card {...args} />;

export const Num = Template.bind({});
Num.args = {
  point: 0,
  clickHandler
}

export const Str = Template.bind({});
Str.args = {
  point: '?',
  clickHandler
}

export const Emoji = Template.bind({});
Emoji.args = {
  point: 'â˜•',
  clickHandler
}