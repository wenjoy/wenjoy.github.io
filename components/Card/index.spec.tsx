import { render } from '@testing-library/react'
import Card from '.'

describe('Card', () => {
    it('should render the card', () => {
        const { container } = render(<Card />)
        expect(container).toMatchSnapshot()
    })

    it('should display number rightly', () => {
        const { container } = render(<Card point={5} />)
        expect(container).toMatchSnapshot()
    })

    it('should display string rightly', () => {
        const { container } = render(<Card point="?" />)
        expect(container).toMatchSnapshot()
    })

    it('should display emoji rightly', () => {
        const { container } = render(<Card point="â˜•" />)
        expect(container).toMatchSnapshot()
    })
})