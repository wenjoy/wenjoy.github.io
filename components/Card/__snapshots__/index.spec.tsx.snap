// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should display emoji rightly 1`] = `
<div>
  <div
    class="card"
  />
</div>
`;

exports[`Card should display number rightly 1`] = `
<div>
  <div
    class="card"
  >
    5
  </div>
</div>
`;

exports[`Card should display string rightly 1`] = `
<div>
  <div
    class="card"
  >
    ?
  </div>
</div>
`;

exports[`Card should render the card 1`] = `
<div>
  <div
    class="card"
  >
    0
  </div>
</div>
`;
